zebra-scan/tests/scanner.rs

```diff
@@ -0,0 +1,257 @@
+    let mut zebrad = test_dir.spawn_child(
+        args!["-c", config_file.clone().to_str().unwrap(), "start"],
+        true,
+    )?;
+    zebrad.expect_stdout_line_matches("Opened Zebra state cache at .*")?;
+
+    // Kill the node now that we have a valid zebra node cache.
+    zebrad.kill(false)?;
+    let output = zebrad.wait_with_output()?;
+
+    // Make sure the command was killed
+    output.assert_was_killed()?;
+
+    // Create a new directory for the scanner
+    let test_dir = testdir()?;
+
+    // Create arguments for the scanner
+    let key = serde_json::json!({"key": ZECPAGES_SAPLING_VIEWING_KEY}).to_string();
+    let listen_addr = "127.0.0.1:8231";
+    let rpc_listen_addr = "127.0.0.1:8232";
+    let zebrad_cache_dir = config.state.cache_dir.clone();
+    let scanning_cache_dir = test_dir.path().join("scanner").to_path_buf();
+    let args = args![
+        "--zebrad-cache-dir",
+        zebrad_cache_dir.to_str().unwrap(),
+        "--scanning-cache-dir",
+        scanning_cache_dir.to_str().unwrap(),
+        "--sapling-keys-to-scan",
+        key,
+        "--listen-addr",
+        listen_addr,
+        "--zebra-rpc-listen-addr",
+        rpc_listen_addr
+    ];
+
+    // Start the scaner using another test directory.
+    let mut zebra_scanner = test_dir.spawn_child(args, false)?;
+
+    // Check scanner was started.
+    zebra_scanner.expect_stdout_line_matches("loaded Zebra scanner cache")?;
+
+    // Check if the grpc server is up
+    let mut client = ScannerClient::connect(format!("http://{listen_addr}")).await?;
+    let request = tonic::Request::new(Empty {});
+    client.get_info(request).await?;
+
+    // Look for 2 scanner notices indicating we are below sapling activation.
+    zebra_scanner.expect_stdout_line_matches("scanner is waiting for Sapling activation. Current tip: [0-9]{1,4}, Sapling activation: 419200")?;
+    zebra_scanner.expect_stdout_line_matches("scanner is waiting for Sapling activation. Current tip: [0-9]{1,4}, Sapling activation: 419200")?;
+
+    // Kill the scanner.
+    zebra_scanner.kill(true)?;
+
+    // Check that scan task started and the scanning is done.
+    let output = zebra_scanner.wait_with_output()?;
+
+    // Make sure the command was killed
+    output.assert_was_killed()?;
+    output.assert_failure()?;
+
+    Ok(())
+}
+
+/// Test that the scanner can continue scanning where it was left when zebrad restarts.
+///
+/// Needs a cache state close to the tip. A possible way to run it locally is:
+///
+/// export ZEBRA_CACHED_STATE_DIR="/path/to/zebra/state"
+/// cargo test scan_start_where_left -- --ignored --nocapture
+///
+/// The test will run zebrad with a key to scan, scan the first few blocks after sapling and then stops.
+/// Then it will restart zebrad and check that it resumes scanning where it was left.
+#[ignore]
+#[tokio::test]
+#[cfg(not(target_os = "windows"))]
+async fn scan_start_where_left() -> Result<()> {
```

Comment:
It wasn't running this test in the `Test All` job before because it doesn't include any optional features like `shielded-scan`, we could avoid running this there with an environment variable, a feature flag, or if there aren't any ignored tests that should be running in the `Test All` job, we could remove the `--include-ignored` flag in `entrypoint.sh` for `RUN_ALL_TESTS`.